### DISCUSSION TITLE
Can we have a REST_OF_THE_OBJECTS class in yolov5 ?

### DISCUSSION THREAD

**Discussion by Hisan-007** [ultralytics/yolov5#10591] [Q&A]
I'm working on a project to detect SKUs of our concern from a group of products bought off the market. 
Most of the items are the ones that are in our list and we have to detect these and therefore annotated in our dataset, but a recent requirement suggest to also detect the ones that are not in our list as UNRECOGNIZED class. 

Initially I decided to convert all the annotations to a single object annotation and run detections but that didn't work quite well. That's because in many images there were those unrecognized items and they didn't have any bounding box  so they were more or less like a background image for the new object model. So what i got in the end was the same results as the previous model only this time the labels were just a single class, I wasn't getting any detections for those unrecognized items. 

Now the unrecognized items can be 40-50 different products approximately. 

MY QUESTION is what would be the best way to detect those unrecognized items 

1. Annotate those unrecognized items as unrecognized class, (i'm not sure how that will work out & whether it would impact the other objects cause the those unrecognized items are quite different from each other in terms of appearance) and the items that we tracking often have subtle differences.
2. do step 1 and then convert all these annotations to the single object class annotations and then train the model.
3. What would you suggest ?

Both of my options require reannotation which is not quite ideal but I'm looking to the expert advice here. If there could be a way to get this done without having to re-annotate it that would be great. 

What do you think would be the best way to do it ?

@glenn-jocher 

Sample of the image
![IMG_2476](https://user-images.githubusercontent.com/76820944/209513522-a15b3270-2d64-4b1e-a549-f9f363336e59.jpeg)

**Comment by Hisan-007** [comment]
any help / ideas would be appreciated.

**Comment by FrancescoSaverioZuppichini** [comment]
I don't see why annotating these objects as UNRECOGNIZED would fail, however, is not very scalable (aka every new class you add requires a retrain). 

It looks like your candy bars have a label written on them, do you think it can be used to find out the right class?
